srcpkgs/jose/template

```diff
@@ -0,0 +1,25 @@
+# Template file for 'jose'
+pkgname=jose
+version=14
+revision=1
+build_style=meson
+configure_args="-Dwerror=false"
+hostmakedepends="asciidoc pkg-config"
+makedepends="jansson-devel openssl-devel zlib-devel"
+short_desc="C-language implementation of Javascript Object Signing and Encryption"
+maintainer="Johannes Heimansberg <git@jhe.dedyn.io>"
+license="Apache-2.0"
+homepage="https://github.com/latchset/jose"
+distfiles="https://github.com/latchset/jose/releases/download/v$version/jose-$version.tar.xz"
+checksum=cee329ef9fce97c4c025604a8d237092f619aaa9f6d35fdf9d8c9052bc1ff95b
+make_check="no" # fails on some architectures
```

Comment:
fails how? on which architectures?

Reply:
Test 2 of 25 dies with a SIGSEGV on x86_64-glibc.

```
 2/25 alg_comp     FAIL              0.05s   killed by signal 11 SIGSEGV
```

Reply:
there's probably a more detailed log in the build dir, but it would be good to fix

Reply:
Agreed, fixing the issue with the failing test would be good. There is a log file under `masterdir/builddir/jose-14/build/meson-logs/testlog.txt` which contains a block of text for each of the tests, for the one failing test it looks like this:

```
==================================== 2/25 ====================================
test:         alg_comp
start time:   18:26:04
duration:     0.05s
result:       killed by signal 11 SIGSEGV
command:      UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 MALLOC_PERTURB_=92 LD_LIBRARY_PATH=/builddir/jose-14/build/lib ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 /builddir/jose-14/build/tests/alg_comp
==============================================================================
```

Not really all that much helpful information as far as I can tell. In the same directory there is also a file called `testlog.junit.xml` which mentions the failing test, but the only thing it contains about that test is:

```
<testcase name="alg_comp" classname="jose" time="0.050262451171875"><failure /></testcase>
```
Looking at the source of the test (`masterdir/builddir/jose-14/tests/alg_comp.c`), it looks like this is supposed to do a couple of compression and decompression tests.

Reply:
Interestingly, if I manually build `jose` and then manually run the tests via `meson test` in my build directory, all tests pass, including the `alg_comp` test. So what I did was:

```
cd masterdir/builddir/jose-14
mkdir testbuild
cd testbuild
meson setup .. --prefix=/usr
ninja
meson test
```

So it appears that something in the build environment that `xbps-src` configures, causes that test to fail.

Reply:
hm, can you try doing `xbps-src chroot` and running the test manually?

Reply:
I just did that with a clean new build directory and the tests all passed again.

Reply:
Ok, so I figured it out. The default meson build-style template uses the `-Db_lto=true`, which appears to be the culprit here. Without it, the problematic test passes without issue.

Reply:
I was wondering, is there an easy way to have the tests run in a debug configuration, so assertions can be enabled when running tests, but disabled for the release build?

Reply:
i don't think so